README.md

```diff
@@ -160,21 +143,104 @@ try {
 }
 ```
 
+Some kinds of dependency fallbacks may be handled by [import maps](https://github.com/wicg/import-maps), but even with the support of import maps, [some scenarios](https://github.com/WICG/import-maps/blob/master/README.md#alternating-logic-based-on-the-presence-of-a-built-in-module) which are outside of the declaratively handled set would still benefit from top-level `await`.
+
+### WebAssembly Modules
+
+WebAssembly Modules are "compiled" and "instantiated" in a logically asynchronous way, based on their imports: Some WebAssembly implementations do nontrivial work at either phase, which is important to be able to shunt off into another thread. To integrate with the JavaScript module system, they will need to do the equivalent of a top-level await. See the [WebAssembly ESM integration proposal](https://github.com/webassembly/esm-integration) for more details.
+
+## Semantics as desugaring
+
+Top-level `await` makes importing a module automatically block on any top-level `await`s. You can think of it like, each module exports a Promise, and after all the `import` statements, but before the rest of the module, the Promises are all `await`ed:
```

Comment:
This seems very spooky-action-at-a-distance to me. I can add an `await` in my module, and alter the execution order of all the modules importing me?

Reply:
This sounds like an argument against the feature as a whole, rather than about Variant A vs Variant B. Did TC39 fail to consider this properly when proposing for Stage 2? (I encourage you to file an issue about your concerns; they are more likely to be lost here.)

Reply:
Yes, this problem was brought to my attention right after the stage 2 meeting, and since it's never been brought back to the committee, and since #42 never happened, i've not had a chance to air my concern fully. Had this occurred to me prior to that meeting, I'd have objected to stage 2.

Reply:
Is it documented clearly in any particular issue?

Reply:
Nope; I think it's the sort of thing that needs talking through before I'm able to verbalize it in an issue well.

Reply:
I would strongly suggest raising an issue with a concrete example, as I don't understand the concern as raised. I think an issue with a textual example will be much clearer than trying to do so verbally.